src/Recommendations/RuleModelRepository.php

```diff
@@ -112,14 +123,54 @@ public function getAll(RuleModel $ruleModel)
         return $this->mapAll($prepared->fetchAll());
     }
 
+    public function getOne(string $id, string $type)
+    {
+        return $this->map($this->get(new RuleModel($id, $type)));
+    }
+
     public function get(RuleModel $ruleModel)
     {
-        $prepared = $this->db->prepare('SELECT Ru.rule_id FROM '.$this->ruleTableName.' as Ru WHERE Ru.id = ? AND Ru.type = ? LIMIT 1;');
-        $prepared->bindValue(1, $ruleModel->getId());
-        $prepared->bindValue(2, $ruleModel->getType());
+        if ($ruleModel->getType() === 'article') {
+            $prepared = $this->db->prepare('
+              SELECT Ru.rule_id, Ru.type, Ru.id, Ru.published, Ru.isSynthetic
+              FROM '.$this->ruleTableName.' as Ru 
+              WHERE Ru.id = ? 
+              AND Ru.type IN (
+                "correction",
+                "editorial",
+                "feature",
+                "insight",
+                "research-advance",
+                "research-article",
+                "research-exchange",
+                "retraction",
+                "registered-report",
+                "replication-study",
+                "short-report",
+                "tools-resources"
```

Comment:
This now allows us to get the type along with the ID assigned by recommendations when a user request comes in, the very top of the bugs chain.